app/javascript/greeting/greetingSlice.js

```diff
@@ -0,0 +1,35 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const initialState = {
+    greeting: {},
+    isLoading: false,
+    error: undefined,
+};
+
+export const takeGreeting = createAsyncThunk('greeting/random', async () => {
+    const greeting = await axios.get('http://localhost:3000/random');
+    // console.log(greeting.data);
+    return await greeting.data;
+});

```

Comment:
- [ ] 🚧 You did a great job working on this milestone! 💯 However, there's an important catch to consider in your code:
  - Using the absolute path `http://localhost:3000/random` leads to a CORS error:
  
![image](https://github.com/emhamza/my_new_rails_app/assets/98453979/612b2616-a48c-455c-9b68-8cc30a42e8bf)

- [ ] Instead, try to use a relative one `/random`. This is because since you're only using one app, it means that you are using the same domain for both of your ends. Fetch will automatically take the client's domain, in this case, `http://localhost:3000`  , and prepend it to the relative endpoint, making it easier to handle your requests without the need for CORS! 🤓 🔧

 - The resulting code would look something like this:

```
export const takeGreeting = createAsyncThunk('greeting/random', async () => {
  const greeting = await axios.get('/random');
  
  return await greeting.data;
});

```

- [ ] You may also consider removing commented code to keep your codebase clean

Reply:
Thank you @frank1738 🌮 
I really appreciate your feedback, it just taught me a new concept. Cheers 🥳